### PR TITLE
common: add atoi(), atol()

### PR DESCRIPTION
**Pull request by asiekierka**

### CODE REVIEW

**Comment by mlund** [comment]
Thanks for this very quick reply! It looks good; the only thing I can think of is if there could be some overlap reduction with the internal version here? https://github.com/llvm-mos/llvm-mos-sdk/blob/666870714f215acd892642557cd1a4a2b6943af9/mos-platform/common/c/printf.c#L187

**Comment by asiekierka** [comment]
Yes, but `printf.c`'s `_atoi` is not compliant with the C standard. The `atoi` in `stdlib` is a superset.

File: mos-platform/common/c/atoi.cc [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	T result = 0;
+	bool sign = false;
+
+	while (isspace(*str)) { str++; }
+
+	if (*str == '+') { str++; }
+	else if (*str == '-') { sign = true; str++; }
+
+	while (*str >= '0' && *str <= '9') {
+		result = (result * 10) + (*(str++));
```
Comment (mysterymath) [inline mos-platform/common/c/atoi.cc:14]: Shouldn't this be `(*(str++) - '0')`?

**Review by mysterymath — APPROVED** [review]

File: mos-platform/common/c/atoi.cc (hunk 1 above)
Comment (asiekierka) [reply mos-platform/common/c/atoi.cc]: Nice catch!

Made sure to *actually* verify if it works this time...

**Pull request merged**